### PR TITLE
CDAP-5108 Metadata audit publish tests

### PR DESCRIPTION
**Pull request by poornachandra**
Build - http://builds.cask.co/browse/CDAP-DUT3686-5

### CODE REVIEW

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/store/MetadataStoreTest.java [hunk 1]
```diff
@@ -198,10 +206,16 @@ public static void setup() throws IOException {
...
     store = injector.getInstance(MetadataStore.class);
+    auditPublisher = injector.getInstance(InMemoryAuditPublisher.class);
+    // Clear dataset creation audit messages
+    auditPublisher.popMessages();
   }
 
   @Test
   public void testPublishing() throws InterruptedException {
+    // clear audit messages
+    auditPublisher.popMessages();
```
Comment (bdmogal) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/store/MetadataStoreTest.java]: why is this needed twice? If its needed at the beginning of every test perhaps put it in an ``@Before`` instead of ``@BeforeClass``?

**Comment by bdmogal** [comment]
@poornachandra is there a way to turn audit publishing off? If so, perhaps we should update the other test in this class too?

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/store/MetadataStoreTest.java (hunk 1 above)
Comment (poornachandra) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/store/MetadataStoreTest.java]: Good point. Moved it to `@Before`

**Comment by poornachandra** [comment]
When audit publishing is off, we don't bind the audit publisher. All tests where audit publishing is not enabled explicitly run in this mode.

**Comment by poornachandra** [comment]
@bdmogal Addressed comments, can you take another look?

**Comment by bdmogal** [comment]
:+1:

**Comment by poornachandra** [comment]
Thanks for the review @bdmogal. Build is green, merging.
